Framework/Algorithms/src/WorkflowAlgorithmRunner.cpp

```diff
@@ -0,0 +1,266 @@
+          algorithm->setProperty(propertyName, value);
+        } else if (valueType == typeid(int)) {
+          const auto &value = propertyTable->cell<int>(row, col);
+          algorithm->setProperty(propertyName, static_cast<long>(value));
+        } else if (valueType == typeid(size_t)) {
+          const auto &value = propertyTable->cell<size_t>(row, col);
+          algorithm->setProperty(propertyName, value);
+        } else if (valueType == typeid(float) || valueType == typeid(double)) {
+          const auto &value = propertyTable->cell<double>(row, col);
+          algorithm->setProperty(propertyName, value);
+        } else if (valueType == typeid(bool)) {
+          const auto &value = propertyTable->cell<bool>(row, col);
+          algorithm->setProperty(propertyName, value);
+        } else if (valueType == typeid(Kernel::V3D)) {
+          const auto &value = propertyTable->cell<V3D>(row, col);
+          algorithm->setProperty(propertyName, value);
+        } else {
+          throw std::runtime_error("Unimplemented column type in " +
+                                   PropertyNames::SETUP_TABLE + ": " +
+                                   valueType.name() + '.');
+        }
+      } catch (std::invalid_argument &e) {
+        throw std::runtime_error("While setting properties for algorithm " +
+                                 algorithmName + ": " + e.what());
+      }
+    }
+    algorithm->execute();
+    if (!algorithm->isExecuted()) {
+      throw std::runtime_error("Workflow algorithm failed to execute.");
+    }
+    queue.pop_front();
+  }
+}
+
+/**
+ * Appends `currentRow` and the rows it depends on to `queue` if needed.
+ * Additionally, configures `propertyTable` appropriately.
+ * @tparam QUEUE a type which supports iterators and has `emplace_back()`
+ * @tparam MAP a type with `std::map` like iterators
+ * @param[in] setupTable a table defining the property dependencies
+ * @param[in,out] propertyTable a table containing the final property values
+ * @param[in] currentRow the row in `setupTable` to configure
+ * @param[in,out] queue a queue of row indexes in order of execution
+ * @param[in] ioMap input/output property lookup table
+ * @param[in,out] rowsBeingQueued a set of rows currently being configured
+ * @throw std::runtime_error in several errorneous cases
+ */
+template <typename QUEUE, typename MAP>
+void WorkflowAlgorithmRunner::configureRow(
+    ITableWorkspace_sptr setupTable, ITableWorkspace_sptr propertyTable,
+    const size_t currentRow, QUEUE &queue, const MAP &ioMap,
+    std::shared_ptr<std::unordered_set<size_t>> rowsBeingQueued) const {
+  if (currentRow > setupTable->rowCount()) {
+    throw std::runtime_error("Current row " + std::to_string(currentRow) +
+                             " out of task table bounds " +
+                             std::to_string(setupTable->rowCount()) + '.');
+  }
+  if (std::find(queue.cbegin(), queue.cend(), currentRow) != queue.cend()) {
```

Comment:
Indeed. I added a bunch of comments to the method to hopefully make it more comprehensible.